src/connectivity.jl

```diff
@@ -748,3 +760,73 @@ function isgraphical(degs::Vector{<:Integer})
     end
     return true
 end
+
+"""
+    isdigraphical(indegree_sequence, outdegree_sequence)
+
+Check whether the given indegree sequence and outdegree sequence are digraphical, that is whether they can be the indegree and outdegree sequence of a digraph.
+
+### Implementation Notes
+According to Fulkerson-Chen-Anstee theorem, a sequence ``\\{(a_1, b_1), ...,(a_n, b_n)\\}`` (sorted in descending order of a) is graphic iff the sum of vertex degrees is even and  ``\\sum_{i = 1}^{n} a_i = \\sum_{i = 1}^{n} b_i\\}`` and the sequence obeys the property -
+```math
+\\sum_{i=1}^{r} a_i \\leq \\sum_{i=r+1}^n min(r-1,b_i) + \\sum_{i=r+1}^n min(r,b_i)
+```
+for each integer r <= n-1. 
+
+See also: [`isgraphical`](@ref)
+"""
+function isdigraphical(
+    indegree_sequence::AbstractVector{<:Integer},
+    outdegree_sequence::AbstractVector{<:Integer},
+)
+    # Check whether the degree sequences have the same length 
+    length(indegree_sequence) == length(outdegree_sequence) || throw(
+        ArgumentError("The indegree and outdegree sequences must have the same length.")
+    )
+    # Check whether the degree sequence is empty
+    !(isempty(indegree_sequence) && isempty(outdegree_sequence)) || return true
+    # Check whether the degree sequences have only non-negative values
+    all(indegree_sequence .>= 0) || throw(
+        ArgumentError("The indegree sequence must contain non-negative integers only.")
+    )
+    all(outdegree_sequence .>= 0) || throw(
+        ArgumentError("The outdegree sequence must contain non-negative integers only.")
+    )
+
+    n = length(indegree_sequence)
+    n == length(outdegree_sequence) || return false
```

Comment:
We already check this on line 783.

Reply:
We removed the redundant check and moved the definition of the variable `n` to the top of the function.